### DISCUSSION TITLE
Github action combining if condition or writing it in Multiple lines

### DISCUSSION THREAD

**Discussion by uditgaurav** [community/community#25375] [Actions]
<p>Hi all,<br>
I have a very long if condition in Github action workflow. Can it be reduced in a lesser number of condition and if not, how to write it in multiple lines?</p>
<p>The command is:</p>
<pre><code class="lang-auto">
      - name: Running all tests
        if: startsWith(github.event.comment.body, '/run-e2e-all') || startsWith(github.event.comment.body, '/run-e2e-first-test') || startsWith(github.event.comment.body, '/run-e2e-second-test') || startsWith(github.event.comment.body, '/run-e2e-third-test') || startsWith(github.event.comment.body, '/run-e2e-fourth-test') || startsWith(github.event.comment.body, '/run-e2e-fifth-test') || startsWith(github.event.comment.body, '/run-e2e-sixth-test') || startsWith(github.event.comment.body, '/run-e2e-seventh-test') || startsWith(github.event.comment.body, '/run-e2e-eighth-test') || startsWith(github.event.comment.body, '/run-e2e-nineth-test') || startsWith(github.event.comment.body, '/run-e2e-tenth-test')
        run: |
          echo "Hello all"
</code></pre>
<p>Please share if you have any thoughts on this.<br>
Thanks!</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/uditgaurav">@uditgaurav</a>,<br>
You can write like as the following:</p>
<pre><code class="lang-yaml">- name: Running all tests
  if: |
    startsWith(github.event.comment.body, '/run-e2e-all') ||
    startsWith(github.event.comment.body, '/run-e2e-first-test') ||
    startsWith(github.event.comment.body, '/run-e2e-second-test') ||
    startsWith(github.event.comment.body, '/run-e2e-third-test') ||
    startsWith(github.event.comment.body, '/run-e2e-fourth-test') ||
    startsWith(github.event.comment.body, '/run-e2e-fifth-test') ||
    startsWith(github.event.comment.body, '/run-e2e-sixth-test') ||
    startsWith(github.event.comment.body, '/run-e2e-seventh-test') ||
    startsWith(github.event.comment.body, '/run-e2e-eighth-test') ||
    startsWith(github.event.comment.body, '/run-e2e-nineth-test') ||
    startsWith(github.event.comment.body, '/run-e2e-tenth-test')
  run: |
    echo "Hello all"
</code></pre>
<p>In the workflow YAML file, if you want to set value of an option (such as <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#env" rel="nofollow noopener"><strong>env.&lt;VAR_NAME&gt;</strong></a>, <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun" rel="nofollow noopener"><strong>run</strong></a>, <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener"><strong>if</strong></a>) as multi-line text, you can generally use the ‘<strong>|</strong>’ character to declare.</p>

**Reply by Propfend** [reply]
is there some way to have multiple conditions?
```bash
- name: Running all tests
if: startsWith(github.event.comment.body, '/run-e2e-all') ||
    run: echo "Hello all"
if: startsWith(github.event.comment.body, '/run-e3e-all') ||
    run: echo "Hello all"
if: startsWith(github.event.comment.body, '/run-e4e-all') ||
    run: echo "Hello all"
if: startsWith(github.event.comment.body, '/run-e5e-all') ||
    run: echo "Hello all"
```
